certifier-osv.md

```diff
@@ -0,0 +1,197 @@
+
+## Architecture
+
+```
+┌──────────────┐     ┌───────────────┐     ┌────────────┐
+│  SBOM Input  │────>│  OSV Certifier│────>│  OSV API   │
+└──────────────┘     └───────────────┘     └────────────┘
+                            │
+                            ▼
+                    ┌───────────────┐
+                    │  GUAC Graph   │
+                    └───────────────┘
+```
+
+## Key Features
+
+- **Vulnerability Detection**: Scans dependencies using SBOMs and cross-references with OSV's vulnerability catalog
+- **Automated Updates**: Regular synchronization with OSV for current vulnerability data
+- **Comprehensive Reporting**: Structured reports showing vulnerabilities by dependency and severity
+- **SBOM Format Support**: 
+  - [CycloneDX](https://cyclonedx.org) (JSON, XML)
+  - [SPDX](https://spdx.dev) (JSON, YAML)
+
+## Integration Details
+
+### Data Collection Process
+
+1. **Dependency Matching**: 
+   - Parses SBOM files for package information
+   - Extracts package names, versions, and ecosystem identifiers
+   - Validates package metadata format
+
+2. **API Querying**: 
+   - Queries OSV using standardized package identifiers
+   - Batch processing for multiple dependencies
+   - Handles API rate limiting and retries
+
+3. **Data Correlation**: 
+   - Maps vulnerabilities to GUAC's dependency graph
+   - Associates vulnerability data with package versions
+   - Maintains relationship between dependencies
+
+### Supported Package Ecosystems
+
+The OSV Certifier supports vulnerability detection across these verified ecosystems:
+
+| Ecosystem | Identifier Format | Example |
+|-----------|------------------|---------|
+| npm | `pkg:npm/{name}@{version}` | `pkg:npm/lodash@4.17.21` |
+| PyPI | `pkg:pypi/{name}@{version}` | `pkg:pypi/requests@2.28.1` |
+| Maven | `pkg:maven/{group}/{artifact}@{version}` | `pkg:maven/org.apache.logging.log4j/log4j-core@2.14.1` |
+| Go | `pkg:golang/{path}@{version}` | `pkg:golang/golang.org/x/text@v0.3.7` |
+| Cargo | `pkg:cargo/{name}@{version}` | `pkg:cargo/serde@1.0.152` |
+| NuGet | `pkg:nuget/{name}@{version}` | `pkg:nuget/Newtonsoft.Json@13.0.1` |
+
+### Feature Support
+
+#### Supported
+- Public vulnerabilities from OSV's database
+- Dependency version mapping with known vulnerabilities
+- Version range analysis
+- [PURL](https://github.com/package-url/purl-spec) package identification
+- Severity classification using [CVSS](https://www.first.org/cvss/) scores
+
+#### Unsupported
+- Private vulnerability detection
+- Ecosystems not covered by OSV
+- Binary vulnerability scanning
+- Custom vulnerability feeds
+
+### Usage
+
+Basic command syntax:
+```bash
+guac osv-certify --sbom <path> [options]
```

Comment:
this should be `guacone certifier osv` see: https://docs.guac.sh/querying-via-cli/#draw-further-insight-from-osvdev